src/BusinessLayer/I_CommunicationService.h

```diff
@@ -11,7 +11,7 @@ class I_CommunicationService : public QObject
     virtual ~I_CommunicationService() {}
 
     virtual void sendSerialData(const unsigned char* packet, int packetLength) = 0;
-    virtual void sendInternetData(const QByteArray& data) = 0;
+    virtual void sendInternetData(const QString& data) = 0;
```

Comment:
What was the reason for switching to QString? The JSON data is originally coming from QByteArray and it is going to std string (which is also a function in QByteArray). At a glance it seems like this has an unneeded conversion.